kong/plugins/rate-limiting/handler.lua

```diff
@@ -29,6 +29,18 @@ local function get_identifier(conf)
     end
   elseif conf.limit_by == "credential" then
     identifier = ngx.ctx.authenticated_credential and ngx.ctx.authenticated_credential.id
+  elseif conf.limit_by == "key" then
+    identifier = kong.request.get_headers()[conf.key_name] or kong.request.get_query()[conf.key_name]
```

Comment:
[kong.request.get_header()](https://docs.konghq.com/latest/pdk/kong.request/#kong_request_get_header) and [kong.request.get_query_arg()](https://docs.konghq.com/latest/pdk/kong.request/#kong-request-get_query_arg) would be more appropriate here.